File: mobile-engage/src/main/java/com/emarsys/mobileengage/notification/command/CompositeCommand.java
package com.emarsys.mobileengage.notification.command;

import com.emarsys.core.util.Assert;

import java.util.List;

public class CompositeCommand implements Runnable {

    private final List<Runnable> commands;

    public CompositeCommand(List<Runnable> commands) {
        Assert.notNull(commands, "Commands must not be null!");
        Assert.elementsNotNull(commands, "Command elements must not be null!");
        this.commands = commands;
    }

    @Override
    public void run() {
        for (Runnable command : commands) {
            command.run();
        }
    }

    public List<Runnable> getCommands() {
        return commands;
    }
}

File: mobile-engage/src/main/java/com/emarsys/mobileengage/notification/command/CompositeCommand.kt
package com.emarsys.mobileengage.notification.command

class CompositeCommand(val commands: List<Runnable>) : Runnable {

    override fun run() {
        val filterIsInstance = commands.filterIsInstance<LaunchApplicationCommand>()
        filterIsInstance.forEach {
            it.run()
        }
        (commands - filterIsInstance).forEach {
            it.run()
        }
    }
}